packages/vite/src/node/plugins/css.ts

```diff
@@ -664,38 +664,47 @@ async function compileCSS(
 interface PostCSSConfigResult {
   options: Postcss.ProcessOptions
   plugins: Postcss.Plugin[]
+  file?: string
 }
 
-let cachedPostcssConfig: PostCSSConfigResult | null | undefined
+let cachedPostcssConfigPath: string | undefined | null
 
 async function resolvePostcssConfig(
+  file: string,
   config: ResolvedConfig
 ): Promise<PostCSSConfigResult | null> {
-  if (cachedPostcssConfig !== undefined) {
-    return cachedPostcssConfig
-  }
-
   // inline postcss config via vite config
   const inlineOptions = config.css?.postcss
+
   if (isObject(inlineOptions)) {
     const result = {
       options: { ...inlineOptions },
       plugins: inlineOptions.plugins || []
     }
     delete result.options.plugins
-    return (cachedPostcssConfig = result)
+
+    return result
+  }
+
+  if (cachedPostcssConfigPath === null) {
+    return null
   }
 
   try {
     const searchPath =
-      typeof inlineOptions === 'string' ? inlineOptions : config.root
-    // @ts-ignore
-    return (cachedPostcssConfig = await postcssrc({}, searchPath))
+      cachedPostcssConfigPath ??
+      (typeof inlineOptions === 'string' ? inlineOptions : config.root)
+
+    const ctx = { file, env: config.mode }
+    const result = (await postcssrc(ctx, searchPath)) as PostCSSConfigResult
```

Comment:
This is obviously very inefficient - I think a better way to do this is have two cached & resolved config values - one for SugarSS and one for normal CSS.

Reply:
Another option is to provide a `cache` function like Babel does it:
https://babeljs.io/docs/en/config-files#apicache